raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGrouper.java

```diff
@@ -1,26 +1,110 @@
+import static org.hypertrace.core.rawspansgrouper.RawSpanGrouperConstants.OUTPUT_TOPIC_PRODUCER;
+import static org.hypertrace.core.rawspansgrouper.RawSpanGrouperConstants.RAW_SPANS_GROUPER_JOB_CONFIG;
+import static org.hypertrace.core.rawspansgrouper.RawSpanGrouperConstants.TRACE_EMIT_TRIGGER_STORE;
+
 import com.typesafe.config.Config;
-import org.hypertrace.core.serviceframework.background.PlatformBackgroundJob;
-import org.hypertrace.core.serviceframework.background.PlatformBackgroundService;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+import org.apache.kafka.streams.state.internals.ValueAndTimestampSerde;
+import org.hypertrace.core.datamodel.RawSpan;
+import org.hypertrace.core.datamodel.RawSpans;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.kafkastreams.framework.KafkaStreamsApp;
 import org.hypertrace.core.serviceframework.config.ConfigClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class RawSpansGrouper extends PlatformBackgroundService {
-  private static Logger LOGGER = LoggerFactory.getLogger(RawSpansGrouper.class);
+public class RawSpansGrouper extends KafkaStreamsApp {
+
+  private static final Logger logger = LoggerFactory
+      .getLogger(RawSpansGrouper.class);
 
   public RawSpansGrouper(ConfigClient configClient) {
     super(configClient);
   }
 
+  public StreamsBuilder buildTopology(Map<String, Object> properties, StreamsBuilder streamsBuilder,
+      Map<String, KStream<?, ?>> inputStreams) {
+    Config jobConfig = getJobConfig(properties);
+    String inputTopic = jobConfig.getString(INPUT_TOPIC_CONFIG_KEY);
+    String outputTopic = jobConfig.getString(OUTPUT_TOPIC_CONFIG_KEY);
+
+    KStream<String, RawSpan> inputStream = (KStream<String, RawSpan>) inputStreams.get(inputTopic);
+    if (inputStream == null) {
+      inputStream = streamsBuilder
+          // read the input topic
+          .stream(inputTopic, Consumed.with(Serdes.String(), null));
+      inputStreams.put(inputTopic, inputStream);
+    }
+
+    // Retrieve the default value serde defined in config and use it
+    Serde valueSerde = defaultValueSerde(properties);
+    StoreBuilder<KeyValueStore<String, ValueAndTimestamp<RawSpans>>> traceStoreBuilder = Stores
+        .keyValueStoreBuilder(Stores.persistentKeyValueStore(INFLIGHT_TRACE_STORE),
```

Comment:
As we saw yesterday, inside `Stores.keyValueStoreBuilder` the default value of `enableCaching` is false. Don't we want to enable caching for the state stores?

Reply:
fixed